### ISSUE TITLE
Unable to Login with Deezer account on Windows 10

### ISSUE THREAD

**Issue by carlonmm** [Superhackman/deezloader-remaster-updated#6]
I was able to login with my account upon installation. I saved my username and password.

When I opened the program a 2nd time, the saved password won't work. I tried retyping my username and password, and even tried changing my password, but it won't work anymore.

**Comment by DnjunaDeep** [comment]
It was normal when I used I some hours ago. But i when I tried to use it a moment ago, I I cannot login either. Also I tried changing my password, it didn't work either.

**Comment by EchoSeven** [comment]
It seems like Deezer changed something regarding the login validation(?). Can't get a "success" via CURL either anymore. It shows "error" no matter what parameters I change.

**Comment by virtualviolet** [comment]
yep, was working fine a few hours ago.... The game of cat and mouse begins again I guess

**Comment by gfiorini** [comment]
Same here, so sad, yesterday it worked and now i'm stuck with login error. Maybe they are changing the APIs ?

**Comment by gfiorini** [comment]
I managed to get it working:

it's an ugly workaround, substitute in your deezer-api.js the init function with this:

Deezer.prototype.init = function(username, password, callback) {
	var self = this;
	NRrequest.post({url: "https://connect.deezer.com/ajax/action.php", headers: this.httpHeaders, form: {type:'login',mail:username,password:password}, jar: true}, (function(err, res, body) {
		if(err || res.statusCode != 200) {
			callback(new Error("Unable to load deezer.com"));
		}else {
			request.get({url: "https://www.deezer.com/", headers: this.httpHeaders, jar: true}, (function(err, res, body) {
				
			var regex = new RegExp(/((?!"api_key\\":\\").*(?=\\"))/g);
			var _token = regex.exec(body);

			const token = "frAIkVj7dAXeGdhbkY8PuyCrKft0eHj3eCwmRoc0";
			self.apiQueries.api_token = token;
			
			const userId = "905099XXX"
			self.userId = userId;

			callback(null, null);


			}).bind(self));
		}
	}));
}

and retrieve your token with another app ( i used https://github.com/social-computing/jmi-deezer, it's a static html webapp which retrieve the access_token)

the userid can be retrieved using https://developers.deezer.com/api/explorer, tab user

then substitute the token and userid variable with the values found

**Comment by Wenthworth** [comment]
Yeah, DeezLoader does not work any more.

**Comment by dr2018sa** [comment]
DeezLoader dont work for me

**Comment by evanrad749** [comment]
Why can't I log in?? I even sign up for a new deezer account. but deezloader won't let me sign no matter what i do! please helpp :(((

**Comment by virtualviolet** [comment]
They have changed the API, but a patch is in circulation... just search redit

**Comment by gfiorini** [comment]
Have you tried it ? Yesterday I manage to get an access_token and override the login process, but anyway the download didnt work for me. Don't know if i messed up something or what. Virtualviolet I looked for a patch on reddit but i couldnt find. can you give the search keys that you used in reddit ?
thanks a lot

**Comment by carlonmm** [comment]
"Deezer api" should do the trick. It's a fairly new thread. The fix works
for me on 4.1.2.

There's also a 5.0.0 ALPHA, but downloading with it seems buggy.

**Comment by carlonmm** [comment]
"Deezer api" should do the trick. It's a fairly new thread. The fix described works for me on 4.1.2.

There's also a 5.0.0 ALPHA, but downloading with it seems buggy.

**Comment by carlonmm** [comment]
"Deezer api" should do the trick. It's a fairly new thread. The fix works for me on 4.1.2.

There's also a 5.0.0 ALPHA, but downloading with it seems buggy.

**Comment by carlonmm** [comment]
"Deezer api" should do the trick. It's a fairly new thread. The fix works for me on 4.1.2.

There's also a 5.0.0 ALPHA, but downloading with it seems buggy.

**Comment by carlonmm** [comment]
(See email reply immediately above)

**Comment by gfiorini** [comment]
thank you carlonmm tonight i will try. 

I report the fix found on reddit:

Deezer.prototype.init = function(username, password, callback) {
	var self = this;
	NRrequest.get({url: self.apiUrl, headers: self.httpHeaders, qs: Object.assign({method:"deezer.getUserData"}, self.apiQueries), json: true, jar: true}, (function(err, res, body) {
		if(!err && res.statusCode == 200) {
			self.apiQueries.api_token = body.results.checkForm;
			NRrequest.post({url: "https://www.deezer.com/ajax/action.php", headers: this.httpHeaders, form: {type:'login',mail:username,password:password,checkFormLogin:body.results.checkFormLogin}, jar: true}, (function(err, res, body) {
				if(err || res.statusCode != 200) {
					callback(new Error("Unable to load deezer.com"));
				}else if(body.indexOf("success") > -1){
					callback(null,null);
				}else{
					callback(new Error("Incorrect email or password."));
				}
			}));
		} else {
			callback(new Error("Unable to load deezer.com"));
		}
	}).bind(self));
}

**Comment by virtualviolet** [comment]
I can't believe git hub doesn't have a user to user pm system...

**Comment by Wenthworth** [comment]
In version 5.0.0 do not download covers.

**Comment by DnjunaDeep** [comment]
The thread in Reddit seems works. Thanks a lot.

**Comment by virtualviolet** [comment]
No worries, I managed to get 3.1.1 working, but on Linux ;-) It actually
seems a lot more stable than the windows version was...

**Comment by xZangetsuxX** [comment]
I cant login too can someone help me psl ?

**Comment by Wenthworth** [comment]
Do not download blocked tracks, in version 4.1.2 they downloaded.
